src/lib/server/shop/payments/stripeMethods.ts

```diff
@@ -135,3 +139,27 @@ export const ensurePaymentIntentState = async (
   }
   return [intent, canRetryPayment];
 };
+
+export const refundConsumable = async (
+  stripeIntentId: string,
+  amount: number,
+) => {
+  try {
+    const intent = await getPaymentIntent(stripeIntentId, {
+      expand: ["latest_charge"],
+    });
+    if (intent.status !== "succeeded") return; // refund can be seen as not necessary
+    // if already refunded, or disputed and lost
+    if ((intent.latest_charge as Stripe.Charge | null)?.refunded) return; // already refunded
```

Comment:
Maybe throw some kind of error here so that we know when were trying to refund something already refunded?

Reply:
I don't agree. Usually, this would be the case if one of two things happened:
1. The admin pressed the refund button before, but for some reason the server crashed AFTER refunding the stripe payment but BEFORE removing the consumable from our database.
2. Someone refunded the user on the stripe dashboard, and THEN went to our dsek admin page to refund the user.

In both cases, I think just pretending like the refund went through is preferable.